### PR TITLE
デバイス内のCalendarの予定を取得する

### PR DESCRIPTION
**Pull request by Hunachi**

### CODE REVIEW

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/CalendarViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class CalendarViewModel : ViewModel() {
+
+    private val useCase: SendCalendarEventUseCase = SendCalendarEventUseCaseImpl()
+
+    fun sendCalendarEvent(context: Context) {
+        try {
+            viewModelScope.launch {
+                useCase.execute(CalendarHelper.setupCalender(context))
+            }
+        }catch (e: Exception){
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/CalendarViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/3067a996df98dd226886069766038df5040996c7/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/CalendarViewModel.kt#L20"></span>Missing spacing after “}”</td>
    </tr>
  </tbody>
</table>

Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/CalendarViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/3067a996df98dd226886069766038df5040996c7/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/CalendarViewModel.kt#L20"></span>Missing spacing before “{“</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarEventRepositoryImpl.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package jp.hacks.smartbread.ui.main.wake.calendar
+
+import com.github.kittinunf.fuel.Fuel
+import com.squareup.moshi.KotlinJsonAdapterFactory
+import com.squareup.moshi.Moshi
+import jp.hacks.smartbread.ui.main.wake.calendar.model.CalendarEvent
+
+internal class CalendarEventRepositoryImpl: CalendarEventRepository {
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarEventRepositoryImpl.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/3067a996df98dd226886069766038df5040996c7/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarEventRepositoryImpl.kt#L8"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarEventRepositoryImpl.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    private val requestAdapter by lazy {
+        Moshi.Builder().add(KotlinJsonAdapterFactory()).build().adapter(CalendarEvent::class.java)
+    }
+
+    override suspend fun sendCalendarEvent(event: CalendarEvent){
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarEventRepositoryImpl.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/3067a996df98dd226886069766038df5040996c7/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarEventRepositoryImpl.kt#L14"></span>Missing spacing before “{“</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarHelper.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package jp.hacks.smartbread.ui.main.wake.calendar
+
+import android.content.Context
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarHelper.kt:3]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarHelper.kt#L3"></span>Imports must be ordered in lexicographic order without any empty lines in-between</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarHelper.kt [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import android.icu.util.Calendar
+import android.text.format.DateFormat
+import jp.hacks.smartbread.ui.main.wake.calendar.model.CalendarEvent
+import me.everything.providers.android.calendar.CalendarProvider
+import me.everything.providers.android.calendar.Event
+import java.sql.Date
+
+
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarHelper.kt:11]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/CalendarHelper.kt#L11"></span>Needless blank line(s)</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/SendCalendarEventUseCase.kt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package jp.hacks.smartbread.ui.main.wake.calendar
+
+import jp.hacks.smartbread.ui.main.wake.calendar.model.CalendarEvent
+
+internal interface SendCalendarEventUseCase{
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/SendCalendarEventUseCase.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/3067a996df98dd226886069766038df5040996c7/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/SendCalendarEventUseCase.kt#L5"></span>Missing spacing before “{“</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/SendCalendarEventUseCaseImpl.kt [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package jp.hacks.smartbread.ui.main.wake.calendar
+
+import jp.hacks.smartbread.ui.main.wake.calendar.model.CalendarEvent
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.withContext
+
+internal class SendCalendarEventUseCaseImpl: SendCalendarEventUseCase {
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/SendCalendarEventUseCaseImpl.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/3067a996df98dd226886069766038df5040996c7/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/SendCalendarEventUseCaseImpl.kt#L8"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/SendCalendarEventUseCaseImpl.kt [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    private val repository: CalendarEventRepository = CalendarEventRepositoryImpl()
+
+    override suspend fun execute(event: CalendarEvent) = coroutineScope {
+        withContext(Dispatchers.IO){
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/SendCalendarEventUseCaseImpl.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/3067a996df98dd226886069766038df5040996c7/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/SendCalendarEventUseCaseImpl.kt#L12"></span>Missing spacing before “{“</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/model/CalendarEvent.kt [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package jp.hacks.smartbread.ui.main.wake.calendar.model
+
+import android.util.Log
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/model/CalendarEvent.kt:3]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/model/CalendarEvent.kt#L3"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/model/CalendarEvent.kt [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+
+data class CalendarEvent (
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/model/CalendarEvent.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/3067a996df98dd226886069766038df5040996c7/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/calendar/model/CalendarEvent.kt#L5"></span>Unexpected spacing before “(“</td>
    </tr>
  </tbody>
</table>

**Comment by fumiya-kume** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          13 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/GoogleFitApiDebugActivity.kt#L7">android/Smartbread/app/src/main/java/jp/hacks/smartbread/GoogleFitApiDebugActivity.kt#L7</a> - Unused import</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L3">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L3</a> - Imports must be ordered in lexicographic order without any empty lines in-between</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L13">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L13</a> - Unused import</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L14">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L14</a> - Unused import</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L15">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L15</a> - Unused import</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L16">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L16</a> - Unused import</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L17">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L17</a> - Unused import</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L18">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L18</a> - Unused import</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L19">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L19</a> - Unused import</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L81">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L81</a> - Unexpected blank line(s) before “}”</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L82">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L82</a> - Needless blank line(s)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/health/HealthPermissionUseCaseImpl.kt#L3">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/health/HealthPermissionUseCaseImpl.kt#L3</a> - Unused import</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/8764882e3d17b395e83d19389070e8c7310793f3/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/BurnBreadUsecaseImpl.kt#L63">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/wake/BurnBreadUsecaseImpl.kt#L63</a> - Needless blank line(s)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by Hunachi** [comment]
時間ないのでマージしちゃいます。すみません。

**Pull request merged**